File: sub-module-hub-service/src/main/java/org/github/auth/shiro/mapper/IShiroMapper.java
package org.github.auth.shiro.mapper;

import java.util.List;
import javax.annotation.Nullable;
import lombok.*;
import org.github.auth.shiro.model.UserDTO;
import org.github.mybatis.MyBatisMapper;
import org.github.shiro.AuthorInfo;

@MyBatisMapper
public interface IShiroMapper {
  @NonNull List<AuthorInfo> queryAuthorInfo(Long userId);

  @Nullable UserDTO queryUserInfo(String username);
}

File: sub-module-hub-service/src/main/java/org/github/auth/shiro/mapper/IShiroMapper.kt
package org.github.auth.shiro.mapper

import org.github.mybatis.MyBatisMapper
import org.github.shiro.AuthorInfo
import org.github.auth.shiro.model.UserDTO

@MyBatisMapper
interface IShiroMapper {
  fun queryAuthorInfo(userId: Long): MutableList<AuthorInfo>

  fun queryUserInfo(username: String): UserDTO?
}
